lxd/cgroup/init.go

```diff
@@ -0,0 +1,74 @@
+package cgroup
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/lxc/lxd/shared/logger"
+)
+
+var cgControllers = map[string]Backend{}
+
+func init() {
+	// Go through the list of resource controllers for LXD.
+	selfCg, err := os.Open("/proc/self/cgroup")
+	if err != nil {
+		if os.IsNotExist(err) {
+			logger.Warnf("System doesn't appear to support CGroups")
+		} else {
+			logger.Errorf("Unable to load list of cgroups: %v", err)
```

Comment:
Technically "Unable to load list of controllers" but it's fine since this file mixes "cgroups" (even "filesystem") and "controllers" since v2 got introduced anyway.